lib/telegraf/rack.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+require 'rack'
+
+module Telegraf
+  # Telegraf::Rack
+  #
+  # This rack middleware collects request metrics and sends them to the telegraf
+  # agent. A `Point` data structure is added to the Rack environment to assign
+  # custom tags and values. This point can be accessed using the environment key
+  # in `::Telegraf::Rack::FIELD_NAME`.
+  #
+  # Example:
+  #
+  #     if (point = request.env[::Telegraf::Rack::FIELD_NAME])
+  #       point.tags[:tag] = 'tag'
+  #       point.values[:value] = 10
+  #     end
+  #
+  #
+  # Tags:
+  #
+  # * `status`:
+  #     Response status unless request errored
+  #
+  #
+  # Values:
+  #
+  # * `request_ms`:
+  #     Total request processing time including response sending.
+  #
+  # * `app_ms`:
+  #     Total application processing time.
+  #
+  # * `send_ms`:
+  #     Time took to send the response body.
+  #
+  # * `queue_ms`:
+  #     Queue time calculated from a `X-Request-Start` header if present. The
+  #     header is expected to be formatted like this `t=<timestamp>` and
+  #     contain a floating point timestamp in seconds.
+  #
+  class Rack
+    FIELD_NAME = 'telegraf.rack.point'
+    HEADER_REGEX = /t=(\d+(\.\d+)?)/.freeze
+
+    Point = Struct.new(:tags, :values)
+
+    def initialize(app, agent:, series: 'rack', tags: {}, logger: nil)
+      @app = app
+      @tags = tags.freeze
+      @agent = agent
+      @series = series.to_s.freeze
+      @logger = logger || ::Logger.new(STDOUT)
```

Comment:
The way it is used `STDERR` seems more appropriate.

Reply:
Changed this to use `rack.logger` from environment by default.